pyro/contrib/epidemiology/compartmental.py

```diff
@@ -868,3 +849,77 @@ def init(self, state):
     def step(self, state):
         with poutine.block(hide_types=["observe"]):
             super().step(state.copy())
+
+
+class _HaarSplitReparam:
```

Comment:
@fehiepsi I found this class provides useful helpers user <--> aux, in addition to our usual `.reparam()` functionality. These extra helpers seem necessary for `poutine.reparam()` to play well with `init_to_value()`. That is, if we reparameterize a model, then a user's custom `init_to_value()` function will need to be converted to different coordinates. WDYT of generalizing these to be part of the `Reparam` interface, say as a pair of methods (`.aux_to_user()`, `.user_to_aux()`) or similar? (in follow-up PRs to Pyro and NumPyro).

Reply:
> These extra helpers seem necessary for poutine.reparam() to play well with init_to_value().

I see. I agree that those helper methods would simplify much of the user code for this init strategy.